internal/service/s3/bucket_lifecycle_configuration.go

```diff
@@ -0,0 +1,390 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceBucketLifecycleConfiguration() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceBucketLifecycleConfigurationCreate,
+		ReadContext:   resourceBucketLifecycleConfigurationRead,
+		UpdateContext: resourceBucketLifecycleConfigurationUpdate,
+		DeleteContext: resourceBucketLifecycleConfigurationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"bucket": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 63),
+			},
+
+			"expected_bucket_owner": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidAccountID,
+			},
+
+			"rule": {
+				Type:     schema.TypeSet,
```

Comment:
Order might actually be important...at least cloud control thinks so https://github.com/hashicorp/terraform-provider-awscc/blob/main/internal/aws/s3/bucket_resource_gen.go#L1267